x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -955,9 +955,9 @@ class AgentPolicyService {
     await soClient.delete(SAVED_OBJECT_TYPE, id);
     await this.triggerAgentPolicyUpdatedEvent(soClient, esClient, 'deleted', id);
 
-    if (options?.removeFleetServerDocuments) {
-      await this.deleteFleetServerPoliciesForPolicyId(esClient, id);
-    }
+    // cleanup .fleet-policies docs on delete
+    await this.deleteFleetServerPoliciesForPolicyId(esClient, id);
```

Comment:
I think that you can also remove `removeFleetServerDocuments` from https://github.com/elastic/kibana/blob/df2a9bba7fbd0f45c747e50e0c400edf6f56e539/x-pack/plugins/fleet/server/services/agent_policy.ts#L893 
This flag seems to be only called in https://github.com/elastic/kibana/blob/0f6c834a26d91aaa942fc41dfe17eb13c34e2bd8/x-pack/plugins/fleet/server/services/preconfiguration/reset_agent_policies.ts#L194

But with your change this line will be always executed so you can remove the flag everywhere.

Reply:
good catch, removed